libs/extensions/src/settings/settings.controller.ts

```diff
@@ -1,16 +1,24 @@
-import { Body, Controller, Get, Param, Patch, Post } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Get,
+  Param,
+  Patch,
+  Post,
+  Query,
+} from '@nestjs/common';
 import { ApiTags } from '@nestjs/swagger';
-import { CreateSettingDto, UpdateSettingDto } from '../dtos';
+import { CreateSettingDto, ListSettingDto, UpdateSettngsDto } from '../dtos';
 import { SettingsService } from './settings.service';
 
 @Controller('settings')
 @ApiTags('Settings')
 export class SettingsController {
-  constructor(private readonly settingsService: SettingsService) { }
+  constructor(private readonly settingsService: SettingsService) {}
 
-  @Get('')
-  listPublic() {
-    return this.settingsService.listPublic();
+  @Get('public')
```

Comment:
Remove `public` from the API endpoints.
List API should list all the Public and private settings but privated settings value should not be revealed.  Just send 'protected' as value in private settings

there shoul also be querys to filter out 'private' 'public' 'readonly' settings

Reply:
cc: @santosh-rumsan

Reply:
@manjik-rumsan, public api  is removed from  the end points. Now we  have an api for listing all settings and private  settings value  and required value can not be revealed.
Just send as "PROTECTED" in  value and required fields.